extensions/amp-base-carousel/1.0/test/test-amp-base-carousel.js

```diff
@@ -143,5 +157,53 @@ describes.realWin(
       expect(renderedSlides[1]).to.equal(userSuppliedChildren[0]);
       expect(renderedSlides[2]).to.equal(userSuppliedChildren[1]);
     });
+
+    describe.only('imperative api', () => {
+      let scroller;
+
+      beforeEach(async () => {
+        const userSuppliedChildren = setSlides(3);
+        userSuppliedChildren.forEach((child) => element.appendChild(child));
+        win.document.body.appendChild(element);
+        await getSlidesFromShadow();
+
+        scroller = element.shadowRoot.querySelector('[class*=scrollContainer]');
+      });
+
+      function invocation(method, args = {}) {
+        const source = null;
+        const caller = null;
+        const event = null;
+        const trust = ActionTrust.DEFAULT;
+        return new ActionInvocation(
+          element,
+          method,
+          args,
+          source,
+          caller,
+          event,
+          trust
+        );
+      }
+
+      it('should execute next and prev actions', async () => {
```

Comment:
This gets called as part of `gulp unit` right? Do we have any guidelines about how long unit tests should take? On a failure, this will always take ~200ms, which seems slow for a unit tests.

Reply:
I think overall we've been doing two things:

1. Our unit tests have been becoming a bit more integration-y in nature. This is especially in Bento since most of unit tests execute the component, not just AMP element. However, they are still not the same as the AMP integration tests which are meant to test a bigger wide AMP Runtime.
2. I think it's ok if the failures take a bit longer, as long as the positive run is fast. Failures already take a longer cycle with fix.